cmd/neofs-adm/internal/modules/morph/deploy.go

```diff
@@ -109,24 +110,44 @@ func deployContractCmd(cmd *cobra.Command, args []string) error {
 	emit.Int(w.BinWriter, 3)
 	emit.Opcodes(w.BinWriter, opcode.PACK)
 	emit.AppCallNoArgs(w.BinWriter, callHash, method, callflag.All)
-	emit.Opcodes(w.BinWriter, opcode.DROP) // contract state on stack
-	if !isUpdate {
-		s, err := c.nnsRegisterDomainScript(nnsCs.Hash, cs.Hash, domain)
-		if err != nil {
-			return err
-		}
-		if len(s) != 0 {
-			c.Command.Printf("NNS: Set %s -> %s\n", domain, cs.Hash.StringLE())
-			w.WriteBytes(s)
-		}
+
+	err = c.sendCommitteeTx(w.Bytes(), -1, false)
+	if err != nil {
+		return fmt.Errorf("can't send deploying transaction: %w", err)
+	}
+
+	err = c.awaitTx()
+	if err != nil {
+		return fmt.Errorf("can't wait deploying transaction: %w", err)
+	}
+
+	if isUpdate {
+		return nil
+	}
+
+	// add record to NNS
+	w = io.NewBufBinWriter()
```

Comment:
BufBinWriter can be reused after previous transaction. Use `w.Reset()` to reset it.

Reply:
fixed